### PR TITLE
Enable I2C on MIMXRT1020-EVK

### PR DESCRIPTION
**Pull request by aurel32**
This PR enables I2C support on the MIMXRT1020-EVK board. Tested with the WM8960 codec for LPI2C1 and with an external chip for LPI2C4.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

File: boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts [hunk 1]
```diff
@@ -76,6 +76,20 @@ arduino_serial: &uart2 {};
 
 &i2c4 {
 	status = "ok";
+
+	fxos8700@1f {
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts]: @galak, how do you feel about adding this node if the sensor isn't populated on the board by default? Is there a dts convention for unpopulated board components? I already started down this path with the sensor interrupt gpios in commit ad52bce1d027a23392ab36d50bbfea24d0cfb2c0, but not sure how far we want to go with it. I'm a bit concerned we could confuse people that try to use the sensor and don't realize it's not actually there.

Comment (aurel32) [reply boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts]: I have to admit it confused me until @MaureenHelm told me to check if the sensor was populated on my board. Do we know if this concerns a limited set of boards? The fact sheet clearly list the FXOS8700CQ sensor, however it's not populated on the photos of the boards.

Comment (MaureenHelm) [reply boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts]: > Do we know if this concerns a limited set of boards?

I'm looking into it. The A1 schematic implies that the sensor should be populated, whereas the A3 schematic marks it DNP. I have an A2 board. My rt1060 and rt1064 boards also don't have the sensor populated. It looks like DNP might be the theme for the sensor on newer rt boards, possibly for cost reasons.

Comment (aurel32) [reply boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts]: Thanks for the search. If newer rt boards do not have it, I agree that we should not provide the DTS entry for it. I have therefore pushed a new version of the PR dropping that commit.

**Review by MaureenHelm — APPROVED** [review]
Please rebase to resolve merge conflicts

**Comment by aurel32** [comment]
> Please rebase to resolve merge conflicts

@MaureenHelm I have just rebased this PR.

**Review by galak — APPROVED** [review]

**Pull request merged**